samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -111,6 +111,9 @@ static const char sock_send_usage_str[] =
 	"                                010203040506070809101112\n"
 	"                                01 02 03 04 05 06 07 08 09 10 11 12\n"
 	"                                01020304 05060708 09101112\n"
+	"  -W, --wait_ack            Request a blocking send operation until the data\n"
+	"                            has been sent on-air and acknowledged by the peer,\n"
+	"                            if required by the network protocol.\n"
```

Comment:
Now that I understand the changes better, I guess we should mention here that you should set `-B 1` whenever this `-W` is used.

Reply:
`-B 1` is default, but maybe add a comment that `-B 0` should not be used?

What about the issue with not logging "socket send failed" for non-blocking sockets in `sock_send()`? If this (TCP related) issue is fixed now I think that restriction should be removed, and the send failure will indicate that the `-B 0 -W` combination is not supported (here it will be `EOPNOTSUPP`).

Reply:
I will create a separate PR for enabling "socket send failed" for non-blocking sockets.

Reply:
I don't know if this issue has been fixes or not. Or do you know it?
I recall it was something harder to manage and I don't recall that there would have been attempts to fix it but I'm not sure.

Reply:
I will do some testing and check around, the reference page does not exist.
Let's move this discussion to PR #13902.